samples/go/poke/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/attic-labs/noms/go/dataset"
+	"github.com/attic-labs/noms/go/spec"
+	"github.com/attic-labs/noms/go/types"
+	flag "github.com/juju/gnuflag"
+)
+
+func main() {
+	var outDSStr = flag.String("out-ds-name", "", "output dataset to write to - if empty, defaults to input dataset")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Poke modifies a single value in a noms database.\n\n")
+		fmt.Fprintf(os.Stderr, "Usage: %s [-out-ds-name=<name>] <ds> <path> <new-val>\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  <ds>      : Dataset to modify\n")
+		fmt.Fprintf(os.Stderr, "  <path>    : Path to a value within <ds> to modify\n")
+		fmt.Fprintf(os.Stderr, "  <new-val> : new value for <path>\n\n")
+		fmt.Fprintln(os.Stderr, "Flags:\n")
+		flag.PrintDefaults()
+	}
+
+	flag.Parse(false)
+
+	if flag.NArg() == 0 {
+		flag.Usage()
+		return
+	}
+
+	if flag.NArg() != 3 {
+		fmt.Fprintln(os.Stderr, "Incorrect number of arguments")
+		return
+	}
+
+	inDS, err := spec.GetDataset(flag.Arg(0))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid input dataset '%s': %s\n", flag.Arg(0), err)
+		return
+	}
+
+	inRoot, ok := inDS.MaybeHeadValue()
+	if !ok {
+		fmt.Fprintln(os.Stderr, "Input dataset has no data")
+		return
+	}
+
+	inPath, err := types.ParsePath(flag.Arg(1))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid path '%s': %s\n", flag.Arg(1), err)
+		return
+	}
+
+	oldVal := inPath.Resolve(inRoot)
+	if oldVal == nil {
+		fmt.Fprintf(os.Stderr, "No value at path '%s' - cannot update\n", inPath.String())
+		return
+	}
+
+	val, _, rem, err := types.ParsePathIndex(flag.Arg(2))
```

Comment:
It looks surprising to see path parsing code used here, perhaps comment that it's somewhat of a hack, since it just so happens that path code knows how to parse values.

You know, ideally that stuff would be factored out of path because it does look pretty weird.